### PR TITLE
PrismaCloudITAdmin

### PR DESCRIPTION
**Pull request by Balaji-S-PANW**
## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues 
None. New Integration.

## Description
Initial commit.

## Screenshots
None

## Minimum version of Demisto
- [x] 4.5.0
- [ ] 5.0.0
- [ ] 5.5.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [x] No

## Must have
- [x] Tests
- [x] Documentation 

## Demisto Partner?
- [ ] The title must be in the following format: **[YOUR_PARTNER_ID] short description**
Partner ID yet to be provided

### CODE REVIEW

**Comment by Balaji-S-PANW** [comment]
> [CLA assistant check](https://cla-assistant.io/demisto/content?pullRequest=7815)
> Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/demisto/content?pullRequest=7815) before we can accept your contribution.
> You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/demisto/content?pullRequest=7815) it.

I have signed CLA, but it still showing as not signed. Could you please check on it.

**Comment by glicht** [comment]
> > [CLA assistant check](https://cla-assistant.io/demisto/content?pullRequest=7815)
> > Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/demisto/content?pullRequest=7815) before we can accept your contribution.
> > You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/demisto/content?pullRequest=7815) it.
> 
> I have signed CLA, but it still showing as not signed. Could you please check on it.

@Balaji-S-PANW we see you signed the CLA. Not clear why it is still showing as "not signed". You can ignore this status check for now.

File: Packs/PrismaCloudITAdmin/Integrations/Prisma_Cloud_IT_Admin/Prisma_Cloud_IT_Admin.py [hunk 1]
```diff
@@ -0,0 +1,523 @@
...
+def get_user_command(client, args):
+
+    scim = verify_and_load_scim_data(args.get('scim'))
+    scim_flat_data = map_scim(scim)
+    user_id = scim_flat_data.get('id')
+    username = scim_flat_data.get('userName')
+    email = scim_flat_data.get('email')
+
+    if not (user_id or username or email):
+        raise Exception('You must provide either the id,, email or username of the user')
```
Comment (dantavori) [inline Packs/PrismaCloudITAdmin/Integrations/Prisma_Cloud_IT_Admin/Prisma_Cloud_IT_Admin.py:258]:
```suggestion
        raise Exception('You must provide either the id, email or username of the user')
```

File: Packs/PrismaCloudITAdmin/Integrations/Prisma_Cloud_IT_Admin/Prisma_Cloud_IT_Admin.py [hunk 2]
```diff
@@ -0,0 +1,523 @@
...
+        active = res_json['enabled']
+        generic_iam_context = OutputContext(success=True, iden=res_json.get('email'), email=res_json.get('email'),
+                                            username=username, details=res_json, active=active)
+    elif res.status_code == 400:
+        generic_iam_context = OutputContext(success=False, iden=user_id, username=username, errorCode=404,
+                                            errorMessage="User Not Found", details=res.headers.get('x-redlock-status'))
+    else:
+        generic_iam_context = OutputContext(success=False, iden=user_id, username=username,
+                                            errorCode=res.status_code,
+                                            errorMessage=res.headers.get('x-redlock-status'),
```
Comment (dantavori) [inline Packs/PrismaCloudITAdmin/Integrations/Prisma_Cloud_IT_Admin/Prisma_Cloud_IT_Admin.py:279]: does the `x-redlock-status` hold the error message in this case?

File: Packs/PrismaCloudITAdmin/Integrations/Prisma_Cloud_IT_Admin/Prisma_Cloud_IT_Admin.yml [hunk 3]
```diff
@@ -0,0 +1,268 @@
...
+  - arguments:
+    - description: SCIM content in JSON format
+      name: scim
+      required: true
+    description: Get a user
+    name: get-user
+    outputs:
+    - contextPath: GetUser
+      description: Command context path
+      type: Unknown
```
Comment (dantavori) [inline Packs/PrismaCloudITAdmin/Integrations/Prisma_Cloud_IT_Admin/Prisma_Cloud_IT_Admin.yml:60]: this looks redundant

File: Packs/PrismaCloudITAdmin/Integrations/Prisma_Cloud_IT_Admin/Prisma_Cloud_IT_Admin.yml [hunk 4]
```diff
@@ -0,0 +1,268 @@
...
+    - contextPath: GetUser.active
```
Comment (dantavori) [inline Packs/PrismaCloudITAdmin/Integrations/Prisma_Cloud_IT_Admin/Prisma_Cloud_IT_Admin.yml:61]: for all commands, all context outputs should start with the integration name, e.g `PrismaCloudITAdmin.GetUser.Active`.

File: Packs/PrismaCloudITAdmin/Integrations/Prisma_Cloud_IT_Admin/Pipfile [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+olefile = "==0.46"
+python-dateutil = "==2.8.0"
+pytz = "==2019.2"
+regex = "==2019.8.19"
+requests = "==2.22.0"
+six = "==1.12.0"
+tzlocal = "==2.0.0"
+urllib3 = "==1.25.3"
+PySocks = "==1.7.0"
+PyYAML = "==5.1.2"
```
[the remark this answers is not in the document]
Comment (Balaji-S-PANW) [reply Packs/PrismaCloudITAdmin/Integrations/Prisma_Cloud_IT_Admin/Pipfile:32]: These are all auto generated.  We are not sure about it. But it looks like the file contains the list of packages required for the integration or the demisto-sdk.

File: Packs/PrismaCloudITAdmin/Integrations/Prisma_Cloud_IT_Admin/Prisma_Cloud_IT_Admin.py [hunk 6]
```diff
@@ -0,0 +1,523 @@
...
+    Returns:
+        'ok' if test passed, anything else will fail the test.
+    """
+    args
+    uri = 'user/id'
+    res = client.http_request(method='GET', url_suffix=uri)
+    if res.status_code == 200 or res.status_code == 400:
+        return 'ok', None, None
+    else:
+        raise Exception(f"{res.status_code} - {res.text}")
```
[the remark this answers is not in the document]
Comment (Balaji-S-PANW) [reply Packs/PrismaCloudITAdmin/Integrations/Prisma_Cloud_IT_Admin/Prisma_Cloud_IT_Admin.py]: I have done the changes and pushed the code

File: Packs/PrismaCloudITAdmin/Integrations/Prisma_Cloud_IT_Admin/Prisma_Cloud_IT_Admin.py (hunk 2 above)
Comment (Balaji-S-PANW) [reply Packs/PrismaCloudITAdmin/Integrations/Prisma_Cloud_IT_Admin/Prisma_Cloud_IT_Admin.py:280]: Yes. It will provide the required error message

**Comment by Balaji-S-PANW** [comment]
> The CircleCI build failed again. @Balaji-S-PANW take a look at the build details [here](https://circleci.com/gh/Balaji-S-PANW/content/10?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link) - and try and fix the issues so that we can merge your proposed changes as soon as possible.
> 
> #### Failed Build Steps

I have looked into the circleci build error mesages. All the error message are related generic test scripts which are available under content/Tests folder. Is there anything we need to do from our side to resolve the issues. Please help us, we are new to demisto.

**Comment by Balaji-S-PANW** [comment]
> ⚠️ We detected security issues in this pull request:
> 
> Hard-Coded Secrets (1)
> https://github.com/demisto/content/blob/8e2d35f6e58a2c9a7078b1f761c225dc60ed9c28/Packs/TruSTAR/Integrations/TruSTAR_V2/Pipfile#L32
> 
> More info on how to fix Hard-Coded Secrets in [General](https://docs.guardrails.io/docs/en/vulnerabilities/general/hard-coded_secrets.html?utm_source=ghpr).
> 
> 👉 Go to the [dashboard](https://dashboard.guardrails.io/default/gh/demisto/content/8e2d35f6e58a2c9a7078b1f761c225dc60ed9c28?type=pull) for detailed results.
> 
> 📥 Happy? Share your [feedback](https://guardrails.typeform.com/to/BCgBkB) with us.

This error was not related PrismaCloudITAdmin Integration. I have fetched the latest from demisto/content master repository to  demisto/Balaji-S-PANW  master and PrismaCloudAdmin repository (my forked repository) and pushed the code. 

The error was in some other Integration appears after taking latest, whether I need to do fix these errors in my repository.

**Comment by glicht** [comment]
@Balaji-S-PANW you can ignore the message about secrets

**Comment by Balaji-S-PANW** [comment]
> The CircleCI build failed again. @Balaji-S-PANW take a look at the build details [here](https://circleci.com/gh/Balaji-S-PANW/content/16?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link) - and try and fix the issues so that we can merge your proposed changes as soon as possible.
> 
> #### Failed Build Steps

Could not able to check what exactly failed in circleci. When I try to access the detail, getting the following mesage "This job has been blocked because resource-class:medium+ is not available on your plan. Please upgrade to continue building.".

Attached screenshot for same. 
Circleci error

**Comment by Balaji-S-PANW** [comment]
> > The CircleCI build failed again. @Balaji-S-PANW take a look at the build details [here](https://circleci.com/gh/Balaji-S-PANW/content/16?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link) - and try and fix the issues so that we can merge your proposed changes as soon as possible.
> > #### Failed Build Steps
> 
> Could not able to check what exactly failed in circleci. When I try to access the detail, getting the following mesage "This job has been blocked because resource-class:medium+ is not available on your plan. Please upgrade to continue building.".
> 
> Attached screenshot for same.
> Circleci error

I was Going through the plan in the circleCI account and I was using Free account and for free account I can't access the docker size of Medium+. Can you please add me to the circleci-PANW organization account
circleCI
.

**Comment by dantavori** [comment]
@Balaji-S-PANW in order to fix the Circle CI issues, please do the following:
1. Unfollow the [content project](https://app.circleci.com/pipelines/github/Balaji-S-PANW/content) created under your Circle CI organization.
2. Save the current branch changes, close this PR and delete this fork.
3. Fork the content repository, create a branch with your saved files and create the PR. You should not follow the forked project, as described [here](https://support.circleci.com/hc/en-us/articles/360008097173-Why-aren-t-pull-requests-triggering-jobs-on-my-organization-).

**Comment by sonaaru** [comment]
> @Balaji-S-PANW in order to fix the Circle CI issues, please do the following:
> 
> 1. Unfollow the [content project](https://app.circleci.com/pipelines/github/Balaji-S-PANW/content) created under your Circle CI organization.
> 2. Save the current branch changes, close this PR and delete this fork.
> 3. Fork the content repository, create a branch with your saved files and create the PR. You should not follow the forked project, as described [here](https://support.circleci.com/hc/en-us/articles/360008097173-Why-aren-t-pull-requests-triggering-jobs-on-my-organization-).

@dantavori :
I have Unfollowed the project in Circle CI. And created the new PR 7983 from another repository. Link to PR: https://github.com/demisto/content/pull/7983

**Comment by ShahafBenYakir** [comment]
@dantavori can we close this?

**Pull request closed without merging**